web/src/settings_exports.ts

```diff
@@ -132,6 +139,124 @@ export function populate_exports_table(exports: RealmExport[]): void {
     }
 }
 
+function sort_user_by_name(a: ExportConsent, b: ExportConsent): number {
+    const a_name = people.get_full_name(a.user_id).toLowerCase();
+    const b_name = people.get_full_name(b.user_id).toLowerCase();
+    if (a_name > b_name) {
+        return 1;
+    } else if (a_name === b_name) {
+        return 0;
+    }
+    return -1;
+}
+
+const export_consents = new Map<number, boolean>();
+const queued_export_consents: (ExportConsent | number)[] = [];
+let export_consent_list_widget: ListWidgetType<ExportConsent>;
+let filter_by_consent_dropdown_widget: DropdownWidget;
+const filter_by_consent_options: Option[] = [
+    {
+        unique_id: 0,
+        name: $t({defaultMessage: "Granted"}),
+    },
+    {
+        unique_id: 1,
+        name: $t({defaultMessage: "Not granted"}),
+    },
+];
+
+function get_export_consents_having_consent_value(consent: boolean): ExportConsent[] {
+    const export_consent_list: ExportConsent[] = [];
+    for (const [user_id, consented] of export_consents.entries()) {
+        if (consent === consented) {
+            export_consent_list.push({user_id, consented});
+        }
+    }
+    return export_consent_list;
+}
```

Comment:
Should this function be merging the `export_consents` data with `people.ts` data to return a list that contains `false` for any user whose consent status we don't specifically know?

As discussed on another PR, I think that might be required to correctly show new user accounts as not having consented if we've not received an explicit data value for them, at least until the server sends an explicit event for every newly created user regardless of consent status.